internal/batches/batch_spec_test.go

```diff
@@ -87,4 +90,83 @@ changesetTemplate:
 			t.Fatalf("wrong error. want=%q, have=%q", wantErr, haveErr)
 		}
 	})
+
+	t.Run("uses unsupported conditional exec", func(t *testing.T) {
+		const spec = `
+name: hello-world
+description: Add Hello World to READMEs
+on:
+  - repositoriesMatchingQuery: file:README.md
+steps:
+  - run: echo Hello World | tee -a $(find -name README.md)
+    if: "false"
+    container: alpine:3
+
+changesetTemplate:
+  title: Hello World
+  body: My first batch change!
+  branch: hello-world
+  commit:
+    message: Append Hello World to all README.md files
+  published: false
+`
+
+		_, err := ParseBatchSpec([]byte(spec), FeatureFlags{})
+		if err == nil {
+			t.Fatal("no error returned")
+		}
+
+		wantErr := `1 error occurred:
+	* step 1 in batch spec uses the 'if' attribute for conditional execution, which is not supported in this Sourcegraph version
+
+`
+		haveErr := err.Error()
+		if haveErr != wantErr {
+			t.Fatalf("wrong error. want=%q, have=%q", wantErr, haveErr)
+		}
+	})
+
+	t.Run("parsing if attribute", func(t *testing.T) {
+		const specTemplate = `
+name: hello-world
+description: Add Hello World to READMEs
+on:
+  - repositoriesMatchingQuery: file:README.md
+steps:
+  - run: echo Hello World | tee -a $(find -name README.md)
+    if: %s
+    container: alpine:3
+
+changesetTemplate:
+  title: Hello World
+  body: My first batch change!
+  branch: hello-world
+  commit:
+    message: Append Hello World to all README.md files
+  published: false
+`
+
+		for _, tt := range []struct {
+			raw  string
+			want string
+		}{
+			{raw: `"true"`, want: "true"},
```

Comment:
(Don't add this, I just want to point out how weird yaml is)
```suggestion
			{raw: `"true"`, want: "true"},
			{raw: `yes`, want: "true"},
```

Reply:
Hehe, that actually doesn't work, because we're effectively skipping the YAML parsing for this attribute: if you put in  `yes` the `interface{}` cannot be cast to a `bool`, but is a string.

Reply:
I think the yaml parser converts it to bool already 🤔

Reply:
```go
func TestYamlFoobar(t *testing.T) {
	type person struct {
		Cool interface{} `json:"Cool" yaml:"cool"`
	}

	personSchema := `{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "Person",
  "description": "Person",
  "type": "object",
  "properties": {
    "cool": {
	  "oneOf": [{ "type": "boolean" }, { "type": "string" }]
    }
  }
}`

	data := []byte(`cool: yes`)
	var p person
	if err := yaml.UnmarshalValidate(personSchema, data, &p); err != nil {
		t.Fatal(err)
	}

	switch cool := p.Cool.(type) {
	case bool:
		t.Errorf("is a bool: %t", cool)
	case string:
		t.Errorf("is a string: %s", cool)
	default:
		t.Errorf("is something else: %+v (%T)", cool, cool)
	}
}

```

prints:

```
--- FAIL: TestYamlFoobar (0.00s)
    batch_spec_test.go:203: is a string: yes
FAIL
FAIL    github.com/sourcegraph/src-cli/internal/batches 0.010s
FAIL
```


Not trying to prove you wrong, but just investigating where it falls apart. I think it's because of our custom yamlv3 unmarshalling etc.

Reply:
Strange 🤔 Well, doesn't matter too much 🤷